pkg/sql/catalog/typedesc/type_desc.go

```diff
@@ -513,6 +513,8 @@ func (desc *immutable) ValidateSelf(vea catalog.ValidationErrorAccumulator) {
 		if desc.GetArrayTypeID() != descpb.InvalidID {
 			vea.Report(errors.AssertionFailedf("ALIAS type desc has array type ID %d", desc.GetArrayTypeID()))
 		}
+	case descpb.TypeDescriptor_VIRTUAL_TABLE_RECORD_TYPE:
+		vea.Report(errors.AssertionFailedf("cannot persist type descriptor kind %s", desc.Kind.String()))
```

Comment:
I find this message confusing. For a start we shouldn't assume that validation only happens during descriptor reads and writes. Really if we believe that a type descriptor of this kind should never even be validated in the first place then that's what the message should read, accompanied with an explanation as to its transient nature.

Reply:
I would also prefer for this validation failure to be exercised in a test.